src/app/components/user/UserWorkspaces/PaginatedUserWorkspaces.js

```diff
@@ -0,0 +1,269 @@
+    />;
+    setFlashMessage(errorMessage, 'error');
+  };
+
+  const onSuccess = (team) => {
+    const path = `/${team.slug}/all-items`;
+    browserHistory.push(path);
+  };
+
+  const setCurrentTeam = (team) => {
+    commitMutation(Relay.Store, {
+      mutation: updateUserMutation,
+      variables: {
+        input: {
+          current_team_id: team.dbid,
+        },
+      },
+      onCompleted: ({ error }) => {
+        if (error) {
+          onFailure(error);
+        } else {
+          onSuccess(team);
+        }
+      },
+      onError: onFailure,
+    });
+  };
+
+  const handleCancelCreateTeam = () => {
+    setShowCreateTeamDialog(false);
+  };
+
+  const handlePageChange = (newPage) => {
+    if (newPage > page) {
+      if (relay.hasMore() && !relay.isLoading() && (cursor + pageSize >= teams.length)) {
+        setIsPaginationLoading(true);
+        relay.loadMore(pageSize, () => {
+          setCursor(cursor + pageSize);
+          setIsPaginationLoading(false);
+        });
+      } else if (cursor + pageSize < teams.length) {
+        setCursor(cursor + pageSize);
+      }
+    } else if (cursor - pageSize >= 0) {
+      setCursor(cursor - pageSize);
+    }
+    setPage(newPage);
+  };
+
+  return (
+    <>
+      <SettingsHeader
+        title={
+          <FormattedMessage
+            id="userSettings.workspacesTitle"
+            defaultMessage="Workspaces [{workspacesCount}]"
+            description="Title for user settings area for user workspaces list they are a memeber of"
+            values={{ workspacesCount: numberOfTeams || 0 }}
+          />
+        }
+        actionButton={
+          <ButtonMain
+            theme="brand"
+            size="default"
+            variant="contained"
+            onClick={() => setShowCreateTeamDialog(true)}
+            label={
+              <FormattedMessage id="switchTeams.newTeamLink" defaultMessage="Create" description="Button label for initiating creating a new team workspace" />
+            }
+          />
+        }
+      />
+      { totalCount > pageSize && // only display paginator if there are more than pageSize worth of workspaces overall in the database
+        <Paginator
+          page={page}
+          pageSize={pageSize}
+          numberOfPageResults={cursor + pageSize >= totalCount ? totalCount % pageSize : pageSize}
+          numberOfTotalResults={totalCount}
+          onChangePage={handlePageChange}
+        />
+      }
+      <div className={styles['user-setting-details-wrapper']}>
+        { teams.length ?
+          <div className={styles['user-setting-content-container']}>
+            { isPaginationLoading && <MediasLoading size="medium" theme="grey" variant="inline" /> }
+            <ul className={cx('teams', styles['user-setting-content-list'])}>
+              {teams.slice(cursor, cursor + pageSize).sort((a, b) => a.name.localeCompare(b.name, undefined, { numeric: true, sensitivity: 'base' })).map(team => (
```

Comment:
This is interesting. Does this mean that we're sorting each page by name? I'd rather have this sorted on the backend before pagination, so pages follow the alphabetical order all along from the first to last page.